MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -4988,14 +5012,27 @@ public string GetHwTonemapFilter(EncodingOptions options, string hwTonemapSuffix
                 return swFilterChain;
             }
 
-            // ffmpeg cannot use videotoolbox to scale
-            var swScaleFilter = GetSwScaleFilter(state, options, vidEncoder, inW, inH, threeDFormat, reqW, reqH, reqMaxW, reqMaxH);
-            newfilters.Add(swScaleFilter);
+            if (!supportsHwScale)
+            {
+                var swScaleFilter = GetSwScaleFilter(state, options, vidEncoder, inW, inH, threeDFormat, reqW, reqH, reqMaxW, reqMaxH);
+                newfilters.Add(swScaleFilter);
+            }
 
             // hwupload on videotoolbox encoders can automatically convert AVFrame into its CVPixelBuffer equivalent
             // videotoolbox will automatically convert the CVPixelBuffer to a pixel format the encoder supports, so we don't have to set a pixel format explicitly here
             // This will reduce CPU usage significantly on UHD videos with 10 bit colors because we bypassed the ffmpeg pixel format conversion
             newfilters.Add("hwupload");
+            if (supportsHwScale)
+            {
+                var hwScaleFilter = GetHwScaleFilter("vt", string.Empty, inW, inH, reqW, reqH, reqMaxW, reqMaxH);
+                if (useHwToneMapping)
+                {
+                    hwScaleFilter = string.IsNullOrEmpty(hwScaleFilter) ? "scale_vt=0:0:bt709:bt709:bt709"
+                        : string.Format(CultureInfo.InvariantCulture, hwScaleFilter, ":bt709:bt709:bt709");
+                }
```

Comment:
For hardware filters, it is best to use the option name explicitly.
Because these will go wrong once upstream changes the order of options.

```suggestion
                var hwScaleFilter = GetHwScaleFilter("vt", null, inW, inH, reqW, reqH, reqMaxW, reqMaxH);
                if (useHwToneMapping)
                {
                    var tonemapArgs = "color_matrix=bt709:color_primaries=bt709:color_transfer=bt709";
                    hwScaleFilter = string.IsNullOrEmpty(hwScaleFilter)
                        ? "scale_vt=" + tonemapArgs
                        : hwScaleFilter + ":" + tonemapArgs;
                }
```